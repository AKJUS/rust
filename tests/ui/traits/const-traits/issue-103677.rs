//@[stock] check-pass
//@ revisions: stock cnst
#![cfg_attr(cnst, feature(const_trait_impl))]

const _: fn(&String) = |s| {
    &*s as &str;
    //[cnst]~^ ERROR: the trait bound `String: const Deref` is not satisfied
};

fn main() {}
