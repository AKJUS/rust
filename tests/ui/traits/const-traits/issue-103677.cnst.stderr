error[E0277]: the trait bound `String: const Deref` is not satisfied
  --> $DIR/issue-103677.rs:6:5
   |
LL |     &*s as &str;
   |     ^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
